packages/next/src/utils.ts

```diff
@@ -2265,6 +2265,7 @@ export async function getMiddlewareBundle({
       source.edgeFunctions[edgeFile] = worker.edgeFunction;
       const route = {
         continue: true,
+        override: true,
```

Comment:
What's the purpose of this property exactly??

Reply:
It overrides the sorting done during merging routes after the build, x-ref: https://github.com/vercel/vercel/blob/5a7851a7f7d4fdf3e99119f059fa34b897736780/packages/routing-utils/src/merge.ts#L40-L44

Reply:
I think this will need to take into account the Next.js version so that it doesn't break existing customers.

Something like `override: gte(version, '2.1.6')`

Reply:
It shouldn't break existing customers, I need to investigate this further but I think we may want to move this before the `beforeFiles` rewrites